### PR TITLE
Update for a better Hex release (0.4.1)

### PR DESCRIPTION
**Pull request by yurrriq**
This PR now includes #18 and #19.

---

Use the [LFE Hex package](https://hex.pm/packages/lfe) since it has no dependencies/issues. There is no additional effort involved for users, since `rebar3` has a Hex dependency resolver internally.

Add `pkg_name` so we can keep `'lfe-compile'` without waiting for the Hex team to support (or decide not to support) hyphens:

``` erlang
{pkg_name, rebar3_lfe_compile}
```

~~Bump to [0.4.3](https://hex.pm/packages/rebar3_lfe_compile/0.4.3) to keep versioning sane wrt Hex.~~

### CODE REVIEW

**Comment by yurrriq** [comment]
Note the Travis failures were from me deleting a merged branch before it could try to compile it :smile:

See [this PR's build](https://travis-ci.org/lfe-rebar3/compile/builds/124887890) for an accurate assessment.

File: src/lfe-compile.app.src [hunk 1]
```diff
@@ -1,11 +1,14 @@
 {application, 'lfe-compile',
  [{description, "The LFE rebar3 compiler plugin"},
-  {vsn, "0.4.0"},
+  {vsn, "0.4.3"},
```
Comment (oubiwann) [inline src/lfe-compile.app.src]: The current release is 0.4.0.

Let's change this to 0.4.1.

File: rebar.config [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {erl_opts, []}.
```
Comment (oubiwann) [inline rebar.config]: I don't want to use hex by default, so let's revert the change to this file.

File: rebar.lock [hunk 3]
```diff
@@ -1,4 +1 @@
-[{<<"lfe">>,
-  {git,"https://github.com/rvirding/lfe.git",
-       {ref,"cd96efc3d2a73ecff7f3e2ef3563b73b098e4844"}},
-  0}].
+[{<<"lfe">>,{pkg,<<"lfe">>,<<"1.0.2">>},0}].
```
Comment (oubiwann) [inline rebar.lock:1]: I don't want to use hex by default, so let's revert the change to this file.

Same thing for `rebar.config`.

**Comment by oubiwann** [comment]
Huh, it looks like Github now supports in-PR edits ... so I addressed my own in-line comments and updated files.

**Comment by oubiwann** [comment]
What do you mean "keep versioning sane with hex.pm"?

Why do we need to jump version numbers?

**Comment by yurrriq** [comment]
Well Hex has 0.4.1 0.4.2 and 0.4.3. I'm not sure if it will treat 0.4.0 as the latest since it will be the latest chronologically or as an old version because of the numbers.

File: rebar.lock [hunk 4]
```diff
@@ -1 +1,4 @@
-[{<<"lfe">>,{pkg,<<"lfe">>,<<"1.0.2">>},0}].
+[{<<"lfe">>,
```
Comment (yurrriq) [inline rebar.lock]: We can't do this. Hex packages must have all Hex packages as dependencies. There's also no reason not to us the LFE hex package, as it has no dependencies and involves no additional effort by the user beyond installing `rebar3`.

**Comment by yurrriq** [comment]
Some of your updates break this PR..

File: rebar.config [hunk 5]
```diff
@@ -1,5 +1,5 @@
...
 
 {deps, [
-  {lfe, "1.0.2"}
+  {lfe, {git, "https://github.com/rvirding/lfe.git", {tag, "v1.0.2"}}}
```
Comment (yurrriq) [inline rebar.config]: We can't do this. Hex packages must have all Hex packages as dependencies. There's also no reason not to use the [LFE hex package](https://hex.pm/packages/lfe), as it has no dependencies and involves no additional effort by the user beyond installing `rebar3`.

File: README.md [hunk 6]
```diff
@@ -50,18 +50,22 @@ $ rebar3 compile
...
+```
 
-   %% hex:
-   {lfe-compile, "0.4.3", {pkg, rebar3_lfe_compile}}
+For use with ``rebar3`` and hex.pm packages:
 
-   %% not both!
+```erlang
+{plugins, [
+   {lfe-compile, "0.4.1", {pkg, rebar3_lfe_compile}}
```
Comment (yurrriq) [inline README.md]: :+1: to splitting up the examples, but this introduces a typo.

**Comment by yurrriq** [comment]
Re: versioning, I've reverted the other ones so your version is fine.

**Comment by yurrriq** [comment]
Ok, I've [tagged](https://github.com/lfe-rebar3/compile/releases/tag/0.4.1) and [published](https://hex.pm/packages/rebar3_lfe_compile) [0.4.1](https://github.com/lfe-rebar3/compile/pull/17/commits/fe8808e6bafa3783ca47e755a709818c0b67702a).

File: rebar.lock (hunk 3 above)
Comment (yurrriq) [inline rebar.lock:1]: We can't do this. If you don't want to use Hex, then we can't at all, unless we do some profile voodoo or something. If you're opposed to Hex support, I guess I could maintain a fork/branch that's identical except for the lfe dep spec.
